huaweicloud/services/acceptance/dataarts/data_source_huaweicloud_dataarts_architecture_ds_template_optionals_test.go

```diff
@@ -0,0 +1,93 @@
+package dataarts
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDatasourceTemplateOptionalFields_basic(t *testing.T) {
+	rName := "data.huaweicloud_dataarts_architecture_ds_template_optionals.test"
+	dc := acceptance.InitDataSourceCheck(rName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckDataArtsWorkSpaceID(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceTemplateOptionalFields_basic(),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(rName, "optional_fields.0.fd_name"),
+					resource.TestCheckResourceAttrSet(rName, "optional_fields.0.description"),
+					resource.TestCheckResourceAttrSet(rName, "optional_fields.0.description_en"),
+					resource.TestCheckResourceAttrSet(rName, "optional_fields.0.required"),
+					resource.TestCheckResourceAttrSet(rName, "optional_fields.0.searchable"),
+
+					resource.TestCheckOutput("fd_name_filter_is_useful", "true"),
+					resource.TestCheckOutput("required_filter_is_useful", "true"),
+					resource.TestCheckOutput("searchable_filter_is_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDatasourceTemplateOptionalFields_basic() string {
+	return fmt.Sprintf(`
+data "huaweicloud_dataarts_architecture_ds_template_optionals" "test" {
+  workspace_id = "%[1]s"
+}
+
+data "huaweicloud_dataarts_architecture_ds_template_optionals" "fd_name_filter" {
+  depends_on   = [data.huaweicloud_dataarts_architecture_ds_template_optionals.test]
```

Comment:
This line is unnecessary here, cause the dependencie has been established by field `fd_name`.
Please check the other parts.

Reply:
done